### PR TITLE
35906 Add Breadcrumbs ESLint rule

### LINKED ISSUE #35906: Add component migration functionality for va-breadcrumbs
## Issue Description

Create an ESLint rule for the new `va-breadcrumbs` web component. `<Breadcrumbs>` -> `<va-breadcrumbs>`

For an example ESLint rule, see https://github.com/department-of-veterans-affairs/vets-website/pull/20002

---
## Tasks
- [ ] Add ESLint rule

### PR DESCRIPTION
**Pull request by andresriveratoro**
## Description
Closes https://github.com/department-of-veterans-affairs/va.gov-team/issues/35906

This PR adds an ESLint rule to assist in the migration from `Breadcrumbs` to `va-breadcrumbs`.

## Original issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/35906

## Testing done
- Unit testing
  - `yarn test:unit script/eslint-plugin-va/`
- Local runs of ESLint on select directories to show warnings
- Testing the suggestion integration with VS code

## Screenshots

https://user-images.githubusercontent.com/36863582/153534307-37e54028-6c86-45a1-b411-4301ec67465d.mov

## Acceptance criteria
- [x] Create an ESLint rule for the new va-breadcrumbs web component

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by timwright12 — APPROVED** [review]

**Review by bkjohnson — COMMENTED** [review]
This looks good, but we may want to wait to merge this until we deal with https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/635. Engineers could get annoyed if they follow the linting suggestion only to find out that the new breadcrumbs component doesn't work well with the `react-router` components.

File: script/eslint-plugin-va/rules/prefer-web-component-library.js [hunk 1]
```diff
@@ -63,11 +120,19 @@ module.exports = {
...
+
+        switch (componentName) {
+          case 'Breadcrumbs':
+            breadcrumbsTransformer(context, node);
+            break;
+          case 'Telephone':
+            telephoneTransformer(context, node);
+            break;
+          default:
+            break;
```
Comment (bkjohnson) [inline script/eslint-plugin-va/rules/prefer-web-component-library.js:135]: Nice job on this :+1:

File: script/eslint-plugin-va/rules/prefer-web-component-library.js [hunk 2]
```diff
@@ -48,6 +48,63 @@ const telephoneTransformer = (context, node) => {
...
+  if (
+    !isLibraryComponent
+      .hasOwnProperty(context.getFilename())
+      ?.hasOwnProperty(componentName)
+  ) {
+    updateIsLibraryImportObj(context, componentName);
+  }
+  return isLibraryComponent[context.getFilename()][componentName];
+};
+
```
Comment (bkjohnson) [inline script/eslint-plugin-va/rules/prefer-web-component-library.js:107]: This looks like a cool optimization, but can you add some comments to explain it a bit? Things that might be good to call out:

- What gets stored in `isLibraryComponent`, key and value
- How are we using `isLibraryComponent`

**Review by bkjohnson — COMMENTED** [review]
Can we add a `valid` test case like this for `<Breadcrumbs>`?

https://github.com/department-of-veterans-affairs/vets-website/blob/6e12936de3da81b4de13734d512ea35d828ba2c6/script/eslint-plugin-va/tests/lib/rules/prefer-web-component-library.js#L24-L33

**Review by timwright12 — APPROVED** [review]

**Review by bkjohnson — APPROVED** [review]
Looks good, though we're still working on a solution to the `<Link>` issue.

**Pull request merged**